backend/resources/contact.py

```diff
@@ -1,34 +1,116 @@
+
+    # available variables: name, email, phone, content, base_url
+
+    internal_message_subject = """
+        Email z {base_url} od {name}
+    """
+
+    internal_message_body = """
+        Nowa wiadomość od {name} <{email}>
+        numer telefonu: {phone}
+        Treść:
+        {content}
+    """
+
+    external_message_subject = """
+        [CfP] Witaj!
+    """
+
+    external_message_body = """
+        Cześć,
+        
+        tu załoga Code For Poznań! Cieszymy się, że z nami jesteś!
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.log = current_app.logger
+        self.base_url = current_app.config["BASE_URL"]
+        self.mail_web_server = current_app.config["MAIL_WEB_SERVER"]
+        self.mail_suppress_send = current_app.config["MAIL_SUPPRESS_SEND"]
+
+        self.hello_addr = f"hello@{self.base_url}"
+        self.notify_addr = f"notifications@{self.base_url}"
+
     def post(self):
         try:
             message_schema = MessageSchema()
-            new_msg = message_schema.load(request.json)
+            data = message_schema.load(request.json)
         except ValidationError as err:
             return {"message": err.messages}, HTTPStatus.BAD_REQUEST
-        name, phone, content = new_msg["name"], new_msg["phone"], new_msg["content"]
-        base_url = os.environ.get("BASE_URL", "codeforpoznan.pl")
-
-        msg = Message(
-            subject=f"Email z {base_url} od {name}",
-            sender=new_msg["email"],
-            reply_to=new_msg["email"],
-            recipients=[f"hello@{base_url}"],
-        )
-        msg.body = f"Nowa wiadomość od {name}, nr tel: {phone} \nTreść:\n {content}"
-        mail.send(msg)
+
+        sender_email = data["email"]
+        context = {"base_url": self.base_url, **data}
+
+        try:
+            # mail from someone (via form) to us, send internal message
+            self.send_message(
+                context=context,
+                subject=self.internal_message_subject,
+                body=self.internal_message_body,
+                sender=self.notify_addr,
+                reply_to=self.notify_addr,
+                recipients=[self.hello_addr],
+            )
+
+            # mail from us to someone, send pretty external message
+            self.send_message(
+                context=context,
+                subject=self.external_message_subject,
+                body=self.external_message_body,
+                sender=self.notify_addr,
+                reply_to=self.hello_addr,
+                recipients=[sender_email],
+            )
+
+        except MessageError:
+            return (
+                {"message": "Failed to send contact message"},
+                HTTPStatus.SERVICE_UNAVAILABLE,
+            )
 
         return {"message": "Contact message successfully sent"}, HTTPStatus.OK
+
+    def send_message(self, *, context: dict, subject: str, body: str, **kwargs):
```

Comment:
forces python to accept keyword only arguments.

```python3
#           first variable   second     third
send_message({'context': 1}, "subject", "body")

the above won't work, you have to call it like so:

send_message(context={'context': 1}, subject="subject", body="body")
```

It's useful for enforcing on api user to call the function with explicit arguments because a person who developed that function might foresee that call to the function might not be obvious. See https://github.com/CodeForPoznan/codeforpoznan.pl_v3/pull/375/files/b4f4c26a90c466b4ceb06573af999efb48ecdc2c#diff-ed7a4a56431b7cc5a9fc51246267b6a34500169fb29baf17ca8883351155e6b7R68 for details - if it would be called without named arguments then it'd be very difficult to understand what's going on.

TLDR: it's a syntax / pretty code thing.